extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/predictor/ModelCache.java

```diff
@@ -0,0 +1,92 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor;
+
+import ai.onnxruntime.OrtException;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import lombok.Getter;
+import org.prebid.server.exception.PreBidException;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ModelCache {
+
+    String gcsBucketName;
+
+    String modelPath;
+
+    @Getter
+    Cache<String, OnnxModelRunner> cache;
+
+    Storage storage;
+
+    String onnxModelCacheKeyPrefix;
+
+    ExecutorService executorService;
+
+    AtomicBoolean isFetching;
+
+    public ModelCache(
+            String modelPath,
+            Storage storage,
+            String gcsBucketName,
+            Cache<String, OnnxModelRunner> cache,
+            String onnxModelCacheKeyPrefix) {
+        this.gcsBucketName = gcsBucketName;
+        this.modelPath = modelPath;
+        this.cache = cache;
+        this.storage = storage;
+        this.onnxModelCacheKeyPrefix = onnxModelCacheKeyPrefix;
+        this.executorService = Executors.newCachedThreadPool();
+        this.isFetching = new AtomicBoolean(false);
+    }
+
+    public OnnxModelRunner getModelRunner(String pbuid) {
+        final String cacheKey = onnxModelCacheKeyPrefix + pbuid;
+        final OnnxModelRunner cachedOnnxModelRunner = cache.getIfPresent(cacheKey);
+
+        if (cachedOnnxModelRunner != null) {
+            return cachedOnnxModelRunner;
+        }
+
+        if (isFetching.compareAndSet(false, true)) {
+            CompletableFuture.runAsync(() -> {
+                try {
+                    fetchAndCacheModelRunner(cacheKey);
+                } finally {
+                    isFetching.set(false);
+                }
+            }, executorService);
```

Comment:
running async supresses any exceptions that might be thrown

anyway, my suggestion is using vertx.executeBlocking(), so consider handling possible errors (logging, retrying, throwing through - whatever you think works best for you to troubleshooting)

Reply:
Have wrapped  fetched in `vertx.executeBlocking()` + return Future<OnnxModelRunner> and Future<ThrottlingThresholds> by ModelCache and ThresholdCache